### DISCUSSION TITLE
約物と行頭／連続

### DISCUSSION THREAD

**Discussion by yamahige** [w3c/jlreq-d#70] [General]
約物の詰め（空白の削除）を、行頭に配置する場合や連続する場合について説明しようとしています。そこで、「行頭はどこか？、連続とはどのような場合か？」を明確にしておく必要はないでしょうか？特に字詰め方向（inline direction）にHTMLの要素やCSSの疑似要素が並ぶ場合が不安です:

- 番号付きリスト（列挙、箇条書き）の番号、本体（箇条内容）
- CSSで自動生成されたテキスト（疑似要素）。自動生成は紙にはない概念だと思うので、説明が難しいと思うが…

VivliostyleやChromeやEdgeが行頭と連続する約物の詰めをサポートしてくれたので、サンプルを作ってみました。

- [Vivliostyleで見る](https://vivliostyle.org/viewer/#src=https://yamahige.github.io/study/punctuation_marks_and_line_head/line-punctuation.html)
- [ブラウザーで見る](https://yamahige.github.io/study/punctuation_marks_and_line_head/line-punctuation.html)

**Comment by KobayashiToshi** [comment]
たしかに，何が行頭かは，問題になりますね．

1　段落の先頭行，折返し行頭，セル内の行頭，回り込みの行頭は，わかりやすいし，同じ扱いでいいでしょう．（段落の先頭行は，いくつか方式がある．）

2　最近はあまり見ないが，窓見出しという岩波新書で採用されていた小見出しの処理法があった．1行の見出しの場合は本文は2行，2行にする小見出しでは本文は3行にするのが普通で，要は複数の行頭に枠をとり，そこに小見出しを配置する方法です．この方法では，段落の先頭行（行頭の全角下げも行わない）も，折返し行頭も，括弧の前のアキは削除していた．このような狭い領域では，特に行頭の見た目の位置をそろえたいといことだと思います．

3　自動生成テキストの意味が私にはわからない．もし，自動生成テキストが1の箇所であれば，1と同じ処理ができる必要がある．

4　箇条書きは問題になるでしょう．この箇条書きでも，“Vivliostyle”の処理でよいと思います．番号が付く場合も付かない場合も，本体部分の行頭をそろえるのがよい．つまり，括弧の前の空白は削除する．番号がつく場合，桁数が異なると行頭位置が異なるが，同じ桁数であれば，漢字の行頭に対し，括弧の前の空白は削除する．以下の例のように番号の後ろを和字間隔で空ける場合，番号の後ろの括弧の後ろのアキを削除し，さらに（1）の括弧の前のアキも削除する方法が一般的．
（1）□「ここに注本文
（2）□ここに注本文

5　箇条書きに似ているが，図版や表組のキャプションでも，図番号や表番号の後ろのアキに続く箇所も問題になる．これも箇条書きと同様な扱いでよい．

6　グリッドレイアウトは，どうなんでしょう．グリッドレイアウトでは，何を実現したいかにもよるでしょうが，横組でいえば，縦の文字の並びをそろえたいというのであれば，その場合は，行頭の括弧の前のアキは，削除しないほうがよいということになる．ただ，行送り方向の位置を規定位置にそろえたいという場合はある．しかし，横組でいえば，縦の文字の並びをそろえたいケースはあまりないようにも思う．あれば，特別の場合で，たぶん，括弧や句読点も，漢字と同じ全角（または一定の幅）で扱うということになるでしょう．

**Comment by himorin** [comment]
さすがに隣り合うブロック要素(inline-blockも含め)については除外ではないでしょうか。必要ならmarginを入れろという話もありますし、そもそもinline directionの「文字」配置の議論から外れると思います。
@KobayashiToshi さんの「自動生成テキストの意味が」は、自動挿入されたテキストと読み替えてもらえればわかりますでしょうか。ちとずれますが、箇条書きの番号の数字の両脇に自動で入るカッコとかドットなど。

**Reply by KobayashiToshi** [reply]
行中（1行）に入る図版などの直後は，行頭といわないし，行頭とはならないでしょう．隣に並ぶのは別の行の文字で，必ずしも所定の位置にくるわけではない．ここで問題となる行頭は，別の行に並ぶ文字との行送り方向（Y方向）の先頭をそろえたい，ということが問題です．

**Reply by yamahige** [reply]
@himorin さん、補足をありがとうございます。「自動生成テキスト」を少し長めに説明すると、次のようになるでしょうか？

デジタルテキストでは、大部分は筆者が入力するテキストでも、一部にテキストがルールに基づいて自動挿入される場合がある。それらは、リスト（列挙）の記号・番号、図表のキャプションの先頭に挿入される記号・番号、脚注・文末注などの本体の前に挿入される注番号、参考文献の本体の前に挿入される文献番号、本文に挿入されてこれらの番号を参照する記号・番号などである。これら自動挿入されるテキストの文字は筆者が決めるとは限らず、テキストのスタイルをデザインする他者によって決められることがある。また、括弧やピリオドなど約物を含むことが多い。

**Reply by KobayashiToshi** [reply]
ということは，以下のような問題がでる可能性がある．
1　句読点，括弧には和文用と欧文用がある．そこで，スタイル（他者）の決め方により，筆者（編集者）にとって，気に入らない場合がでる可能性がある．それは入力の際にわかるんだろうか．分かれば，別のスタイルを選べばよいのかもしれない．
2　筆者（編集者）は，スタイルの選択において，縦組なら縦組，横組なら横組を考慮しているが，その形式を読者が変えてしまった場合は，どうなるんだろう．
3　自動生成テキストの番号等の後ろに筆者（編集者）の入力するテキストの先頭に和字間隔がある場合，自動生成テキストの番号等の後ろが和文用の括弧であれば，括弧＋和字間隔の組合せから，自動生成テキストの番号等の後ろの括弧の二分アキは削除されるので問題はでない．これがラテン文字では削除されないが，和文用の括弧の場合と，結果は同じになる．
4　自動生成テキストの場合，筆者（編集者）の入力する行の先頭が漢字や仮名の場合と括弧の場合がある．この場合，箇条書きや図版の番号などでは，たぶん，漢字や仮名の先頭と，括弧の先頭をそろえたい．つまり，括弧の前の二分のアキを削除したいと考えることが多いと思われる．その場合，以下のケースになる．
　4.1　自動生成テキストの番号が①などの場合（つまり，自動生成テキストの番号等に括弧や句読点が付かないで，筆者が入力するテキストとの間がベタの場合）は括弧の前の二分アキが残る．ここを行頭と考えれば，二分アキを削除できる．
　4.2　自動生成テキストの番号などの後ろに和字間隔がある場合，これは和字間隔＋括弧の組合せから二分は削除される．ここを行頭と考えて二分は削除されるとしてもよい．この場合，和字間隔＋括弧の組合せから削除の2つが機能するとも考えられるが，和字間隔＋括弧の組合せから削除は機能しないと思われるので問題はない．これは，自動生成テキストが中点で筆者が入力するテキストとの間がベタの場合も同じになる．
　4.3　“(1)”のように自動生成テキストの番号をくくる括弧（ピリオドでも同じ）がラテン文字用の場合，括弧の連続による二分アキの削除の機能は働かないと思われるので，二分アキを削除できない．3.1と同様に行頭と考えれば，二分アキを削除できる．

というように考えれば，自動生成テキストの後ろは，行頭と考えてよいが，自動生成テキストは，行長に挿入されるケースもある．したがって，“行頭に発生する”自動生成テキストの後ろという限定が必要になる．

**Comment by yamahige** [comment]
> 4　箇条書きは問題になるでしょう．この箇条書きでも，“Vivliostyle”の処理でよいと思います．番号が付く場合も付かない場合も，本体部分の行頭をそろえるのがよい．つまり，括弧の前の空白は削除する．番号がつく場合，桁数が異なると行頭位置が異なるが，同じ桁数であれば，漢字の行頭に対し，括弧の前の空白は削除する．以下の例のように番号の後ろを和字間隔で空ける場合，番号の後ろの括弧の後ろのアキを削除し，さらに（1）の括弧の前のアキも削除する方法が一般的．
（1）□「ここに注本文
（2）□ここに注本文

なるほど、やはりそうなんですね。この件を「箇条書き」という観点から整理して別のdiscussionにしてみました、「[順序付きリストの揃え、挙動、約物](https://github.com/w3c/jlreq-d/discussions/72)」です。

**Comment by himorin** [comment]
整理されて別discussion #72 となっているという宣言があるので閉じる
